matrix/Quaternion.hpp

```diff
@@ -44,14 +63,73 @@ class Quaternion : public Vector<Type, 4>
         q(3) = 0;
     }
 
+    /**
+     * Constructor from Matrix41
+     *
+     * @param other Matrix41 to copy
+     */
     Quaternion(const Matrix41 & other) :
         Vector<Type, 4>(other)
     {
     }
 
+    /**
+     * Constructor from dcm
+     *
+     * Instance is initialized from a dcm representing coordinate transformation
+     * from frame 2 to frame 1.
+     *
+     * @param dcm dcm to set quaternion to
+     */
     Quaternion(const Dcm<Type> & dcm) :
         Vector<Type, 4>()
     {
+        set_from_dcm(dcm);
+    }
+
+    /**
+     * Constructor from euler angles
+     *
+     * This sets the instance to a quaternion representing coordinate transformation from
+     * frame 2 to frame 1 where the rotation from frame 1 to frame 2 is described
+     * by a 3-2-1 intrinsic Tait-Bryan rotation sequence.
+     *
+     * @param euler euler angle instance
+     */
+    Quaternion(const Euler<Type> & euler) :
+        Vector<Type, 4>()
+    {
+        set_from_euler(euler);
+    }
+
+    /**
+     * Constructor from quaternion values
+     *
+     * Instance is initialized from quaternion values representing coordinate
+     * transformation from frame 2 to frame 1.
```

Comment:
why from frame 2 to frame 1? To be consistent it should be 1 to 2.

I see the problem that we still try to explain how the frames are used within the px4 code without calling them by the real names (inertial/body). This only leads to confusion.

The reason this seems necessary is that euler angles describe the reverse transformation from the other representations and frames are switched in the conversion functions.

I wonder if it would make sense to redefine the transformation of dcm and quaternions to have consistent rotation representations?